src/foam/core/PropertyInfo.java

```diff
@@ -20,6 +20,7 @@
 
   public boolean getTransient();
   public boolean getRequired();
+  public Class getPropertyClass();
```

Comment:
Sorry to be pedantic, but I'm not super happy with this name.  What about getValueClass() ?  getPropertyClass sounds like the class of the property, but what we we're returning is the class of the value we store in a given property.